test/lib/clients/autoyast_test.rb

```diff
@@ -32,13 +31,29 @@
       instance_double(Y2Autoinstallation::AutoSequence, run: :next)
     end
 
+    let(:module_map) do
+      {
+        "language" => {
+          "Name"                         => "Language",
+          "Icon"                         => "yast-language",
+          "X-SuSE-YaST-AutoInst"         => "all",
+          "X-SuSE-YaST-AutoInstResource" => "language",
+          "X-SuSE-YaST-Group"            => "System",
+          "X-SuSE-YaST-AutoInstClient"   => "language_auto"
+        }
+      }
+    end
+
     before do
+      # reset singleton
+      allow(Yast::Desktop).to receive(:Modules)
+        .and_return(module_map)
+      Singleton.__init__(Y2Autoinstallation::Entries::Registry)
```

Comment:
It looks rather tricky. You might consider implementing the "singleton" yourself as I am not sure about how stable is this method call (can it change with Ruby 2.x?). On the other hand, I am somehow fine if it is needed in the tests only.

Reply:
well, I found it by googling and works well, so far. But I agree that we should not duplicate it and add test only helper to reset it